File: client/src/models/job_meta.ts
class JobMeta {
  SID!: string
  // Title!: string
  InputJSONSchema?: string

  constructor(props: Partial<JobMeta>) {
    // super()
    for(const prop in props) {
      (this as any)[prop] = (props as any)[prop]
    }
  }

  static deserialize(json: any): JobMeta {
    const job = new JobMeta(json as Partial<JobMeta>)
    return job
  }
}

export default JobMeta

File: src/models/job_meta.go
package models

type JobMeta struct {
	SID             string
	Title           string
	Perform         func(job *Job) error
	InputJSONSchema *string
}

func (jobMeta JobMeta) ToContract() map[string]string {
	res := map[string]string{
		"SID":   jobMeta.SID,
		"Title": jobMeta.Title,
	}
	if jobMeta.InputJSONSchema != nil {
		res["InputJSONSchema"] = *jobMeta.InputJSONSchema
	}
	return res
}
